### PR TITLE
T2D-363 — Validate study accession on creation

### PR DESCRIPTION
**Pull request by tskir**
* Enforce validation of accession; don't allow comma or dot symbols
* Update documentation to clarify that comma is not accepted in accession
* Add a test case for not allowing invalid study accessions

### CODE REVIEW

File: docs/using-the-service.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Using the metadata service
+
+## Study queries
+
+Please note that a comma symbol (`,`) is not permitted to be used in a study accession, because this symbol is used to separate study accessions when configuring access control.
```
Comment (tcezard) [inline docs/using-the-service.md]: We should document the dot as well.
```suggestion
Please note that a comma (`,`) and dot (.) symbols are not permitted to be used in a study accession. Dot is used to separate a study accession and version and comma is used to separate study accessions when configuring access control.
```

**Comment by tskir** [comment]
Rebased the PR on top of master after the merge of #83

File: metadata-ws/src/test/resources/features/study.feature [hunk 2]
```diff
@@ -1011,3 +1011,9 @@ Feature: study object
 
     When I request PATCH STUDY with content {""}
     Then the response code should be 400
+
+
+  Scenario: Trying to create a study with an invalid accession name (containing a comma symbol) must fail
```
Comment (andresfsilva) [inline metadata-ws/src/test/resources/features/study.feature]: It would be good to include another one to check it fails when the accession has a `dot`

Comment (tskir) [reply metadata-ws/src/test/resources/features/study.feature]: Thank you for a suggestion, addressed in https://github.com/EBIvariation/metadata-ws/pull/84/commits/c2233c1301e04bee15af6874c318e29ee1748874

**Pull request merged**